components/login/provider/index.jsx

```diff
@@ -33,217 +33,203 @@ import {
+  const bem = useBem(useStyles());
 
-    try {
-      await login({ ...AUTH_OPTIONS, oidcIssuer: providerIri });
-      setLoginError(null);
-    } catch (error) {
-      if (!e.target.value && !providerIri) {
-        return;
-      }
+  const [loginError, setLoginError] = useState(null);
+  const [providerIri, setProviderIri] = useState(provider?.iri || "");
+  const [inputValue, setInputValue] = useState(provider?.label || "");
 
-      setLoginError(error);
-    }
-  };
-}
+  const loginFieldRef = createRef();
 
-export function setupErrorHandler(setLoginError) {
-  return (error) => {
-    setLoginError(error);
-  };
-}
+  useEffect(() => {
+    loginFieldRef.current?.querySelector("input")?.focus();
+  }, [loginFieldRef]);
 
-export function getErrorMessage(error) {
-  const postFix = " Please fill out a valid Solid Identity Provider.";
+  const handleSubmit = useCallback(
+    async (ev) => {
+      // null if called from "create-option":
+      ev?.preventDefault();
 
-  if (hasError(error, ERROR_REGEXES.INVALID_IDP)) {
-    return "This URL is not a Solid Identity Provider.";
-  }
+      const error = await validateProviderIri(providerIri);
+      setLoginError(translateErrors(error));
 
-  if (hasError(error, ERROR_REGEXES.HANDLER_NOT_FOUND)) {
-    return "Please fill out a valid Solid Identity Provider.";
-  }
+      if (!error) {
+        handleLogin(providerIri);
+      }
+    },
+    [setLoginError, handleLogin, providerIri]
+  );
 
-  return `We were unable to log in with this URL.${postFix}`;
-}
+  const handleInputChange = useCallback(
+    (_ev, newValue) => {
+      setLoginError(null);
 
-export default function Provider({ defaultError, provider }) {
-  const bem = useBem(useStyles());
-  const classes = useStyles();
-  const { login } = useSession();
-  const [loginError, setLoginError] = useState(defaultError);
-  const theme = useTheme();
-  const [providerIri, setProviderIri] = useState(provider?.iri || "");
+      if (!newValue) {
+        setInputValue("");
+        setProviderIri("");
+        return;
+      }
 
-  const loginFieldRef = createRef();
+      let issuer = newValue;
+      if (!issuer.startsWith("http://") && !issuer.startsWith("https://")) {
+        issuer = `https://${issuer}`;
+      }
 
-  useEffect(() => {
-    if (!provider) return;
+      setInputValue(newValue);
+      setProviderIri(issuer);
+    },
+    [setLoginError, setInputValue, setProviderIri]
+  );
+
+  const handleProviderChange = useCallback(
+    (_ev, newValue, reason) => {
+      setLoginError(null);
 
-    loginFieldRef.current?.querySelector("input").focus();
-  }, [provider, loginFieldRef]);
+      if (reason === "blur") {
+        return;
+      }
 
-  const onProviderChange = setupOnProviderChange(setProviderIri, setLoginError);
-  const handleLogin = setupLoginHandler(login, setLoginError);
-  const onError = setupErrorHandler(setLoginError);
-  const providersWithIdp = provider ? [provider, ...providers] : providers;
```

Comment:
I ended up removing the provider given by the `?idp=` parameter from the autocomplete options, as it has no icon and would likely look out of place in the list, and the user is probably just likely to hit "go" or enter when the page loads.